### PR TITLE
New algorithm to load SPICE XML data

### PR DESCRIPTION
**Pull request by wdzhou**
SPICE records 2D anger camera's data in XML format.  The new algorithm is designed to read this type of XML file and create a MatrixWorkspace for it.

### CODE REVIEW

**Comment by wdzhou** [comment]
retest please.

**Comment by wdzhou** [comment]
Please

**Comment by wdzhou** [comment]
test this please

**Comment by wdzhou** [comment]
For tester: A new algorithm LoadSpiceXML2DDet is implemented in this algorithm.  In order to test, you can download any XML files of HFIR's HB3A from http://neutron.ornl.gov/user_data/hb3a/exp355/Datafiles/.  You may run the algorithm to load data and check the output MatrixWorkspace including the spectra and sample logs.

**Comment by wdzhou** [comment]
retest please.

**Comment by wdzhou** [comment]
retest this please

**Comment by wdzhou** [comment]
Jekins, test this please.

**Comment by FedeMPouzols** [comment]
Jenkins, retest this please

File: Code/Mantid/Framework/DataHandling/src/LoadSpiceXML2DDet.cpp [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+      createMatrixWorkspace(vec_xmlnode, numpixelX, numpixelY, detlogname);
+
+  setProperty("OutputWorkspace", outws);
+}
+
+//----------------------------------------------------------------------------------------------
+/** Parse SPICE XML file for one Pt./measurement
+ * @brief LoadSpiceXML2DDet::parseSpiceXML
+ * @param xmlfilename
+ * @param logstringmap
```
Comment (FedeMPouzols) [inline Code/Mantid/Framework/DataHandling/src/LoadSpiceXML2DDet.cpp]: I think `logstringmap` should be `vecspicenode`?

**Comment by FedeMPouzols** [comment]
Code looks very good and it comes with a unit test and doc tests. I think it's ready to be merged in. 

I only spotted a very minor issue with a doxygen tag (see inline comment above). Please have a look at it, and I'll merge this PR. I'd also suggest adding a short description of the parameters if there's any point that needs to be clarified (like expected format, typical usage, etc.).

**Comment by FedeMPouzols** [comment]
Just another comment. I tested the algorithm with a few example XML files from the link that you posted above. Everything seems to work well. Maybe in the future a default value for the 'DetectorGeometry' could be guessed from the file? For interactive use it would be convenient, although I'm not sure if this algorithm will be used directly, or just as part of the HFIR HB3A load/reduce workflow.

**Comment by FedeMPouzols** [comment]
OH and another comment, could you add the ticket number for this?

File: Code/Mantid/Framework/DataHandling/src/LoadSpiceXML2DDet.cpp (hunk 1 above)
Comment (wdzhou) [inline Code/Mantid/Framework/DataHandling/src/LoadSpiceXML2DDet.cpp]: Yes, you are right on this.  I forgot to update the parameter's name after I changed the input from map to vector.  It has been corrected.

**Comment by FedeMPouzols** [comment]
We got a glitch on the win7 build, so I stopped the builds that were still running. Let's try again...

**Comment by FedeMPouzols** [comment]
Jenkins, retest this please

**Comment by FedeMPouzols** [comment]
Jenkins, retest this please

**Pull request merged**